### PR TITLE
util: Fix gethostname() and freeaddrinfo() calls

### PR DESCRIPTION
**Pull request by liweitianux**
I was passing the wrong size to the `gethostname()` call because the `hostname` is now dynamically allocated and cannot use `sizeof` to get its size.  This mistake causes that the obtained hostname is empty.

The `freeaddrinfo()` was placed in the wrong place.  Fix it.

Fallback to `_POSIX_HOST_NAME_MAX` if `sysconf()` fails.

Always null-terminate the `hostname`.

(Edited to match my updated patch.)

### CODE REVIEW

File: src/util.c [hunk 1]
```diff
@@ -49,16 +53,16 @@ void hostname(char** out)
 
 	if ((result == 0) && (info != NULL))
 	{
-		dot = strchr(info->ai_canonname, '.');
+		char* dot = strchr(info->ai_canonname, '.');
```
Comment (liweitianux) [inline src/util.c]: Do we need to check the return value of `dot`?

**Comment by liweitianux** [comment]
@cylgom Thanks for reviewing.  I'll fix it later (maybe tomorrow).

One question: is there a specific reason that you prefer `type* v` instead of the more common style `type *v`? The latter should be better when you have to write something like `type *v1, *v2`.  Thanks.

Cheers,

**Comment by cylgom** [comment]
That's the one eternal schism of C programmers :) There are arguments on  each side though:

`type *v`
- makes `type *v1, *v2;` more readable and appear logical
- makes function pointers look good but less readable `void (*fun_ptr)(int) = &fun;`
- makes pointer-returning function pointers less readable `int *(*fun_ptr)(int) = &fun;`
- makes it look like a pointer is not a type and thus does not reflect the way compilers work

`type* v`
- makes `type* v1, *v2;` inconsistent, obscure and completely unreadable.
- makes function pointers look bad but more readable `void (* fun_ptr)(int) = &fun;`
- makes pointer-returning function pointers more readable `int* (* fun_ptr)(int) = &fun;`
- makes it look like a pointer is a type and thus reflects the way compilers work

Actually I used to be like you and use the first spacing convention. But one day I realized `type *v1, *v2;` is very little used in practice, precisely because it's counterintuitive to most people. Carefully looking at the arguments for and against each camp, we can clearly see this is the most important one. So `type *v1, *v2;` looks a lot better, granted, but I never write such declarations, so I changed my mind and started using `type* v;`, because all the other minor arguments are in favor of this spelling.

But to me this debate confirms we need a better-designed replacement for the C language.

**Comment by liweitianux** [comment]
Thanks for the detailed explanation. Since you adopt the style, then I'll follow yours.

On the other hand, I think the `hostname` obtained via `gethostname()` is sufficient, so there is no need to bother with the `getaddrinfo()` and simplify this code a lot.  I had a look at LightDM, and it only uses `gethostname()`.  What do you think?

Cheers.

**Comment by cylgom** [comment]
I just tested a few things and it looks like you're right. Let's remove getaddrinfo().

**Comment by liweitianux** [comment]
Hi @cylgom , I've pushed a new commit here to simplify the `hostname()` function.  Please see the commit message for some details.

Cheers.

**Comment by cylgom** [comment]
Thanks a lot!

**Pull request merged**